pandas/core/sorting.py

```diff
@@ -319,6 +319,33 @@ def nargsort(
     return indexer
 
 
+def nargminmax(values, method: str):
+    """
+    Implementation of np.argmin/argmax but for ExtensionArray and which
+    handles missing values.
+
+    Parameters
+    ----------
+    values : ExtensionArray
+    method : {"argmax", "argmin"}
+
+    Returns
+    -------
+    int
+    """
+    assert method in {"argmax", "argmin"}
+    func = np.argmax if method == "argmax" else np.argmin
+
+    mask = np.asarray(isna(values))
```

Comment:
Sparse would like to avoid this `np.asarray` if possible. Will it need to override `argmax` and `argmin`?

If so, not necessary to do to here but can you open an issue if you agree that it should be done that way?

Reply:
Yes, for an efficient implementation, SparseArray would need to override this.

Now, `SparseArray._values_for_argsort` already densifies as well, since SparseArray just uses the base class (`np.array(self)`), so that seems a general area of improvement. So certainly out of scope here I would say, but will open an issue for that.

Reply:
-> https://github.com/pandas-dev/pandas/issues/34197